File: lisca-py/src/lisca_py/cli/commands/movie.py
from __future__ import annotations

from pathlib import Path
from typing import Annotated

import typer

from ...app.movie import run_movie
from ...common.progress import progress_json_stderr


def command(
    workspace: Annotated[Path, typer.Option("--workspace", exists=True, file_okay=False)],
    pos: Annotated[int, typer.Option("--pos")],
    roi: Annotated[int, typer.Option("--roi")],
    channel: Annotated[int, typer.Option("--channel")],
    time: Annotated[str, typer.Option("--time")],
    output: Annotated[Path, typer.Option("--output")],
    fps: Annotated[int, typer.Option("--fps")] = 10,
    colormap: Annotated[str, typer.Option("--colormap")] = "grayscale",
    spots: Annotated[Path | None, typer.Option("--spots", exists=True, dir_okay=False)] = None,
) -> None:
    run_movie(workspace, pos, roi, channel, time, output, fps, colormap, spots, on_progress=progress_json_stderr)

File: lisca-rs/src/cli/commands/movie.rs
use clap::Args;
use serde::{Deserialize, Serialize};

#[derive(Args, Clone, Serialize, Deserialize)]
pub struct MovieArgs {
    #[arg(long)]
    pub workspace: String,
    #[arg(long)]
    pub pos: u32,
    #[arg(long)]
    pub roi: u32,
    #[arg(long)]
    pub channel: u32,
    #[arg(long)]
    pub time: String,
    #[arg(long)]
    pub output: String,
    #[arg(long, default_value_t = 10)]
    pub fps: u32,
    #[arg(long, default_value = "grayscale")]
    pub colormap: String,
    #[arg(long)]
    pub spots: Option<String>,
    #[arg(long, default_value_t = false)]
    pub no_progress: bool,
}
